docs/components/index.md

```diff
@@ -0,0 +1,355 @@
+
+```csharp
+@using System
+@implements IDisposable
+
+...
+
+@functions {
+    public void Dispose()
+    {
+        ...
+    }
+}
+```
+
+## Routing
+
+Routing in Blazor is achieved by providing a route template to each accessible component in the app.
+
+When a *\*.cshtml* file with an `@page` directive is compiled, the generated class is given a [RouteAttribute](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.mvc.routeattribute) specifying the route template. At runtime, the router looks for component classes with a `RouteAttribute` and renders whichever component has a route template that matches the requested URL.
+
+Multiple route templates can be applied to a component. The following component responds to requests for `/BlazorRoute` and `/DifferentBlazorRoute`:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/BlazorRoute.cshtml?start=1&end=4)]
+
+## Route parameters
+
+Blazor components can receive route parameters from the route template provided in the `@page` directive. The Blazor client-side router uses route parameters to populate the corresponding component parameters.
+
+*RouteParameter.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/RouteParameter.cshtml?start=1&end=8)]
+
+Optional parameters aren't supported, so two `@page` directives are applied in the example above. The first permits navigation to the component without a parameter. The second `@page` directive takes the `{text}` route parameter and assigns the value to the `Text` property.
+
+## JavaScript/TypeScript interop
+
+To call JavaScript libraries or custom JavaScript/TypeScript code from .NET, the current approach is to register a named function with JavaScript/TypeScript. Place the `registerFunction` call in the *index.html* file or a JavaScript file (*\*.js*) loaded by the *index.html* file. Place the inline JavaScript or `<script>` tag below `<script type="blazor-boot"></script>`, and the JavaScript/TypeScript loads at the correct time and only executes once. 
+
+```javascript
+Blazor.registerFunction('doPrompt', function(message) {
+    return prompt(message);
+});
+```
+
+Wrap the named function for calls from .NET:
+
+```csharp
+public static bool DoPrompt(string message)
+{
+    return RegisteredFunction.Invoke<bool>("doPrompt", message);
+}
+```
+
+This approach has the benefit of working with JavaScript build tools, such as [webpack](https://webpack.js.org/).
+
+The Mono team is working on a library that exposes standard browser APIs to .NET.
+
+## Razor support
+
+**Razor directives**
+
+Razor directives active with Blazor apps are shown in the following table.
+
+| Directive | Description |
+| --------- | ----------- |
+| [@functions](https://docs.microsoft.com/aspnet/core/mvc/views/razor#functions) | Adds a C# code block to a component. |
```

Comment:
Inconsistent formatting in the "Directive" column. Some of the items are surrounded by backticks; some aren't. I think all of them should be.